File: srcs/requirements/wordpress/create_products.py
import csv
import subprocess
import json

# Replace with your CSV file name
csv_file = "/tmp/lv.csv"

# Replace with the names of the columns you want to use
columns_to_print = ["name_lv", "description_lv", "small_text_lv", "price", "picture", "meta_description_lv"]

site = "lv.bio113-dev.com"

with open(csv_file, mode='r', encoding='utf-8') as file:
    csv_reader = csv.DictReader(file)
    for row in csv_reader:
        # Convert the row to a dictionary using column names
        values = {column: row[column] for column in columns_to_print}

        # For images and meta_data, create an array of objects
        images = json.dumps([{"src": "https://" +  "bio113.com/th/340x300_6/" +  values["picture"].strip('/')}])
        #meta_data = json.dumps([{"key": "description", "value": values["meta_description_lv"]}])

        command = [
            "wp", "--url={}".format(site), "wc", "product", "create",
            "--path={}".format("/var/www/wordpress"),
            "--name={}".format(values["name_lv"]),
            "--type=simple",
            "--description={}".format(values["description_lv"]),
            "--short_description={}".format(values["small_text_lv"]),
            "--regular_price={}".format(values["price"]),
            "--images={}".format(images),
            #"--meta_data={}".format(meta_data),
            "--user={}".format('admin')
        ]

        # Execute the subprocess command
        try:
            subprocess.run(command, check=True)
        except Exception as e:
            command = [
                "wp", "--url={}".format(site), "wc", "product", "create",
                "--path={}".format("/var/www/wordpress"),
                "--name={}".format(values["name_lv"]),
                "--type=simple",
                "--description={}".format(values["description_lv"]),
                "--short_description={}".format(values["small_text_lv"]),
                "--regular_price={}".format(values["price"]),
                #"--meta_data={}".format(meta_data),
                "--user={}".format('admin')
            ]
            print("ERROR:", e)

File: srcs/requirements/wordpress/create_products.php
<?php

// Assuming $argv[1] contains the language code
$lang = $argv[1];
$csvFile = "/tmp/" . $lang . ".csv";

$columnsToPrint = ["name_" . $lang, "description_" . $lang, "small_text_" . $lang, "price", "picture", "meta_description_" . $lang];

$site = $lang . ".bio113-dev.com";

if (($handle = fopen($csvFile, "r")) !== FALSE) {
    $header = fgetcsv($handle); // Assuming the first row contains headers
    $columnsIndex = array_flip($header);

    while (($row = fgetcsv($handle)) !== FALSE) {
        $values = [];
        foreach ($columnsToPrint as $column) {
            // Extracting the required values based on column names
            $values[$column] = $row[$columnsIndex[$column]];
        }

        // For images, creating an array of objects
        $images = json_encode([["src" => "https://" . "bio113.com/th/340x300_6/" . trim($values["picture"], '/')]]);

        $ret = array();
        $command = "wp --url={$site} wc product create " .
                   "--path=/var/www/wordpress " .
                   "--name='" . $values["name_" . $lang] . "' " .
                   "--type=simple " .
                   "--description='" . $values["description_" . $lang] . "' " .
                   "--short_description='" . $values["small_text_" . $lang] . "' " .
                   "--regular_price=" . $values["price"] . " " .
                   "--images='" . $images . "' " .
                   "--user=admin 2>&1";

        exec($command, $ret, $out);
        echo $ret[0] . "\n";
    }
    fclose($handle);
}
?>
